### PR TITLE
We shouldn't be changing decimal places as user type, we should do that on blur.

### PR DESCRIPTION
**Pull request by segun**
Fixes: #12523 

Explanation:  
We are changing decimal place of input as users type. This PR moved that logic from onChange to onBlur

Manual testing steps:  
1. select bsc network;
2. select USDT;
3. select transfer;
4. input address;
5. input amount 1195.00556
6. issue happens;
7. even after I fix the amount, the next page is wrong, too;

Before 
https://user-images.githubusercontent.com/44811/141193249-c0c7fe4b-3bdb-4cf9-9c4a-0db317b31290.mov

After
https://user-images.githubusercontent.com/44811/141193292-e3ffb904-8db9-41ad-8340-958eb204d3cd.mov

### CODE REVIEW

File: ui/components/ui/token-input/token-input.component.js [hunk 1]
```diff
@@ -78,6 +78,26 @@ export default class TokenInput extends PureComponent {
   handleChange = (decimalValue) => {
     const { token: { decimals } = {}, onChange } = this.props;
 
+    const newDecimalValue = decimalValue;
```
Comment (danjm) [inline ui/components/ui/token-input/token-input.component.js:81]: There is a lot of identical code in `handleChange` and `handleBlur`. Perhaps `handleChange` can accept a new property like `applyDecimals` and then `handleBlur` can just call `handleChange`, but pass that parameter in

**Comment by mcmire** [comment]
@segun I'm having trouble reproducing this bug. Would you mind posting two videos,  one demonstrating the behavior before this commit and one demonstrating after?

File: ui/components/ui/token-input/token-input.component.js [hunk 2]
```diff
@@ -75,11 +75,12 @@ export default class TokenInput extends PureComponent {
     return Number(decimalValueString) ? decimalValueString : '';
   }
 
-  handleChange = (decimalValue) => {
+  handleChange = (decimalValue, applyDecimals = false) => {
```
[the remark this answers is not in the document]
Comment (segun) [reply ui/components/ui/token-input/token-input.component.js:78]: yeah, that's fine.

**Comment by segun** [comment]
> @segun I'm having trouble reproducing this bug. Would you mind posting two videos, one demonstrating the behavior before this commit and one demonstrating after?

I've added to the initial comment

**Review by mcmire — DISMISSED** [review]
Thanks for the videos. This looks good to me.

File: ui/components/ui/token-input/token-input.component.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (segun) [reply ui/components/ui/token-input/token-input.component.js:78]: I don't think we should do this in this PR, the UnitInput component is used in several other components and will expand the scope of this PR.

**Review by jpuri — APPROVED** [review]
LGTM

**Review by mcmire — APPROVED** [review]

**Pull request merged**